### PR TITLE
`isequal` fix

### LINKED ISSUE #48: Define `isequal`
```
Using Intervals

julia> NaN == NaN
false

julia> isequal(NaN, NaN)
true

julia> Interval(NaN, NaN) == Interval(NaN, NaN)
false  # This seems like the correct behaviour

julia> isequal(Interval(NaN, NaN), Interval(NaN, NaN))
false  # This should return true according to how `==` vs `isequal` work.
```

### PR DESCRIPTION
**Pull request by rofinn**
Closes #48

### CODE REVIEW

File: test/comparisons.jl [hunk 1]
```diff
@@ -406,6 +406,15 @@ const INTERVAL_TYPES = [Interval, AnchoredInterval{Ending}, AnchoredInterval{Beg
         @test superset([a, b]) == Interval(1, 5, true, true)
     end
 
+    @testset "equal -0.0/0.0" begin
+        a = Interval(0.0, -0.0)
+        b = Interval(-0.0, 0.0)
```
Comment (omus) [inline test/comparisons.jl]: Can you do:
```julia
a = convert(A, Interval(...))
b = convert(B, Interval(...))
```

File: test/comparisons.jl [hunk 2]
```diff
@@ -406,6 +406,15 @@ const INTERVAL_TYPES = [Interval, AnchoredInterval{Ending}, AnchoredInterval{Beg
...
+
+        @test a == b
+        @test !isequal(a, b)
+        @test hash(a) != hash(b)
```
Comment (omus) [inline test/comparisons.jl]: Should add the other comparisons to ensure they work as expected

Comment (rofinn) [reply test/comparisons.jl]: The endpoint comparisons?

Comment (omus) [reply test/comparisons.jl]: The other comparisons which are performed in this file like `<` and so on

**Pull request merged**